common/parse.go

```diff
@@ -23,6 +23,10 @@ func ParseResult(
 	marshalFunc func([]map[string]any, []string) ([]ReadResultRow, error),
 	fields []string,
 ) (*ReadResult, error) {
+	if resp == nil {
+		return &ReadResult{}, nil
```

Comment:
Nice catch! This should probably be an error, because even if the HTTP response body itself is empty, we'd expect it to still be wrapped around by `JSONHTTPResponse`